src/decompression.jl

```diff
@@ -434,35 +434,32 @@ function decompress_single_color!(
 end
 
 function decompress!(
-    A::SparseMatrixCSC{R}, B::AbstractMatrix{R}, result::StarSetColoringResult
-) where {R<:Real}
-    @compat (; S, compressed_indices) = result
-    check_same_pattern(A, S)
-    nzA = nonzeros(A)
-    for k in eachindex(nzA, compressed_indices)
-        nzA[k] = B[compressed_indices[k]]
-    end
-    return A
-end
-
-function decompress!(
-    A::SparseMatrixCSC{R}, B::AbstractMatrix{R}, result::StarSetColoringResult, uplo::Symbol
+    A::SparseMatrixCSC{R},
+    B::AbstractMatrix{R},
+    result::StarSetColoringResult,
+    uplo::Symbol=:F,
 ) where {R<:Real}
     @compat (; S, compressed_indices) = result
-    uplo == :F && check_same_pattern(A, S)
-    rvS = rowvals(S)
     nzA = nonzeros(A)
-    l = 0  # assume A has the same pattern as the triangle
-    for j in axes(S, 2)
-        for k in nzrange(S, j)
-            i = rvS[k]
-            if in_triangle(i, j, uplo)
-                l += 1
-                nzA[l] = B[compressed_indices[k]]
+    if uplo == :F
+        check_same_pattern(A, S)
+        for k in eachindex(nzA, compressed_indices)
+            nzA[k] = B[compressed_indices[k]]
+        end
+    else
+        rvS = rowvals(S)
+        l = 0  # assume A has the same pattern as the triangle
+        for j in axes(S, 2)
+            for k in nzrange(S, j)
+                i = rvS[k]
+                if in_triangle(i, j, uplo)
+                    l += 1
+                    nzA[l] = B[compressed_indices[k]]
+                end
             end
         end
+        @assert l == nnz(A)
     end
-    @assert l == length(nonzeros(A))
```

Comment:
This was here for a very specific reason. I don't rember the exact discussion around it but the gist is that we don't have the guarantee `nnz(A) == nonzeros(A)`, so checking against `length(nonzeros(A))` was safer.

Reply:
We should probably add a comment explaining that though

Reply:
I remember now. Should we remove this `@assert`?

Reply:
I would keep it, it's a guardrail against simple user mistakes which would otherwise silently return the wrong result. Typically if the user passes the whole matrix in SparseMatrixCSC but requires one of the two triangles, the filling will be all wrong